src/dnx.coreclr.unix/tpa.cpp

```diff
@@ -0,0 +1,105 @@
+        pArray[25] = strdup("System.Runtime.Extensions.ni.dll");
+        pArray[26] = strdup("System.Runtime.Handles.ni.dll");
+        pArray[27] = strdup("System.Runtime.InteropServices.ni.dll");
+        pArray[28] = strdup("System.Runtime.Loader.ni.dll");
+        pArray[29] = strdup("System.Text.Encoding.ni.dll");
+        pArray[30] = strdup("System.Text.Encoding.Extensions.ni.dll");
+        pArray[31] = strdup("System.Threading.ni.dll");
+        pArray[32] = strdup("System.Threading.Overlapped.ni.dll");
+        pArray[33] = strdup("System.Threading.Tasks.ni.dll");
+        pArray[34] = strdup("System.Threading.ThreadPool.ni.dll");
+    }
+    else
+    {
+        pArray[0] = strdup("Internal.Runtime.Uri.dll");
+        pArray[1] = strdup("Internal.Uri.dll");
+        pArray[2] = strdup("dnx.coreclr.managed.dll");
+        pArray[3] = strdup("dnx.host.dll");
+        pArray[4] = strdup("Microsoft.Framework.Runtime.Interfaces.dll");
+        pArray[5] = strdup("Microsoft.Framework.Runtime.Loader.dll");
+        pArray[6] = strdup("mscorlib.dll");
+        pArray[7] = strdup("System.dll");
+        pArray[8] = strdup("System.AppContext.dll");
+        pArray[9] = strdup("System.Collections.dll");
+        pArray[10] = strdup("System.Collections.Concurrent.dll");
+        pArray[11] = strdup("System.ComponentModel.dll");
+        pArray[12] = strdup("System.Console.dll");
+        pArray[13] = strdup("System.Core.dll");
+        pArray[14] = strdup("System.Diagnostics.Debug.dll");
+        pArray[15] = strdup("System.Globalization.dll");
+        pArray[16] = strdup("System.IO.dll");
+        pArray[17] = strdup("System.IO.FileSystem.dll");
+        pArray[18] = strdup("System.IO.FileSystem.Primitives.dll");
+        pArray[19] = strdup("System.Linq.dll");
+        pArray[20] = strdup("System.Reflection.dll");
+        pArray[21] = strdup("System.Reflection.Extensions.dll");
+        pArray[22] = strdup("System.Reflection.Primitives.dll");
+        pArray[23] = strdup("System.Resources.ResourceManager.dll");
+        pArray[24] = strdup("System.Runtime.dll");
+        pArray[25] = strdup("System.Runtime.Extensions.dll");
+        pArray[26] = strdup("System.Runtime.Handles.dll");
+        pArray[27] = strdup("System.Runtime.InteropServices.dll");
+        pArray[28] = strdup("System.Runtime.Loader.dll");
+        pArray[29] = strdup("System.Text.Encoding.dll");
+        pArray[30] = strdup("System.Text.Encoding.Extensions.dll");
+        pArray[31] = strdup("System.Threading.dll");
+        pArray[32] = strdup("System.Threading.Overlapped.dll");
+        pArray[33] = strdup("System.Threading.Tasks.dll");
+        pArray[34] = strdup("System.Threading.ThreadPool.dll");
+    }
+
+    *ppNames = pArray;
+    *pcNames = count;
+
+    return true;
+}
+
+BOOL FreeTpaBase(const LPWSTR* values, const size_t count)
+{
+    for (size_t idx = 0; idx < count; ++idx)
+    {
+        delete[] values[idx];
```

Comment:
@ellismg I don't think anything blocking us from doing this. When I wrote these codes only my cat (she stay by my side at the time) and me knows why I use `_strdup`. Now only my cat knows.   JK. I guess one of the reasons was trying to reuse the data between crossgened file names and original names. But end with hard coded all the codes because it helps to allow simplier build scripts.